### DISCUSSION TITLE
접근 제어자들의 차이점

### DISCUSSION THREAD

**Discussion by seunghye218** [2024-woowacourse-study/level-interview#8] [Java]
.

**Comment by gyungchan-jo** [comment]
public: 어떤 클래스에서든 접근할 수 있는 가장 넓은 범위의 접근 제어자입니다.

protected: 동일한 패키지에 속한 클래스들과 해당 클래스를 상속받은 클래스들에게 접근을 허용합니다. 다른 패키지에 속한 클래스에서는 접근할 수 없지만, 해당 클래스를 상속받은 경우에는 접근이 가능합니다.

default: 동일한 패키지에 속한 클래스들에게만 접근을 허용합니다. protected 접근 제어자와 다른 부분은 상속받은 클래스가 다른 패키지에 위치하면 접근을 허용하지 않습니다.

private: 해당 멤버에 대한 접근을 해당 클래스 내부에서만 허용합니다. 다른 클래스에서는 접근할 수 없습니다.

**Comment by le2sky** [comment]
접근 제어자는 클래스와 변수, 메서드, 생성자에 대한 접근을 제한하는 키워드입니다.
자바에서 제공하는 접근 제어자는 4가지가 있으며, 다음과 같은 차이가 있습니다.

private : 클래스 내부에서만 접근 가능합니다.
default : 자바의 기본 접근 제어자이며, 같은 패키지 내의 클래스만 접근할 수 있습니다. package-private으로도 불립니다.
protected : 같은 패키지 내부의 클래스, 혹은 다른 패키지의 서브 클래스에서 접근할 수 있습니다.
public : 모든 클래스에서 접근 가능합니다.

접근 제어자를 적절하게 설정해준다면
클래스의 구현 세부 사항을 숨길 수 있습니다.
또한, 접근 제어자로 다른 개발자들에게 의도를 전달할 수도 있습니다.
